lib/quick-secrets/config.rb

```diff
@@ -4,22 +4,31 @@ module QuickSecrets
   class Configuration
     attr_reader :config
     attr_reader :config_file
-    
-    # TODO add config file as command line argument so it can be passed to here
-    def initialize(config_file = nil)
+
+    CONFIG_DEFAULT = "/etc/quick_secrets/config.yml"
+    CONFIG_TEMPLATE = __dir__+"/templates/config.yml"
+
+    def initialize(args = nil)
+      # default to that provided on the command line
+      @config_file = args['--config']
```

Comment:
I agree with this order of precedence, however it also needs to be placed in the `def []` method in this class

Reply:
Added ability to pull config options out of command line options.  Requires that options be specified in the usage DOCOPT string, but otherwise anything can be captured, similarly to how options are found via environment variables.